### PR TITLE
adding method to deploy template lock contract

### PR DESCRIPTION
**Pull request by julien51**
# Description

This is the first of all PRs to add support for v12.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-js/scripts/compressAbi.js [hunk 1]
```diff
@@ -94,6 +94,8 @@ function formatDeployableBytecode() {
 const bytecode = {
 ${Object.keys(output).map(version => {
   return `  ${version}: {
+    PublicLock:
```
Comment (julien51) [inline unlock-js/scripts/compressAbi.js]: For some reason, this bit was missing previously!

File: unlock-js/src/walletService.js [hunk 2]
```diff
@@ -149,6 +152,32 @@ export default class WalletService extends UnlockService {
...
+  /**
+   * Deploys a new template for locks
+   * It's a regular lock, but it will never work (purchase function fails)
+   * It just used as template whose address is fed into configUnlock to deploy
+   * locks through a proxy (keeping gas prices much lower)
+   * @param {*} version
+   * @param {*} callback
+   */
+  async deployTemplate(version, callback) {
+    const factory = new ethers.ContractFactory(
```
Comment (nfurfaro) [inline unlock-js/src/walletService.js]: @julien51 I wonder if this is where the problem is. From the `ethers` docs, :
> new ethers . ContractFactory ( abi , bytecode [ , signer ] )
Creates a factory for deployment of the Contract with bytecode, and the constructor defined in the abi. The signer will be used to send any deployment transaction.

Our contract(PublicLock.sol) doesn't have a constructor defined. Maybe you could try this with `ethers . ContractFactory . fromSolidity ( compilerOutput [ , signer ] )` instead?

https://docs.ethers.io/ethers.js/html/api-contract.html#deploying-a-contract

Comment (julien51) [reply unlock-js/src/walletService.js]: hum. I am not sure. What's strange is that the issue is transient. Sometime, the test just deploys fine, some other times, it fails.

Comment (julien51) [reply unlock-js/src/walletService.js]: Also, we actually do not have the `.sol` files at our disposal here inside of `unlock-js`.

Comment (nfurfaro) [reply unlock-js/src/walletService.js]: RIght. I was thinking we could use the artifacts(publicLock.json) here, but not sure if those are available?
However, the fact that its transient makes me doubt that this is the issue

**Comment by julien51** [comment]
ok, that was an issue with constructors! All fixed now as we onkly need this for v1.2/5!

PTAL

**Pull request merged**